tasks.py

```diff
@@ -761,3 +798,132 @@ def frontend_build(c):
+    2. invoke frontend-download --tag 0.13.0
+       Downloads the frontend build from the releases.
+    3. invoke frontend-download --file /home/vscode/Downloads/frontend-build.zip
+       This will extract your zip file and place the contents at the correct destination
+    """
+
+    import functools
+    import subprocess
+    from tempfile import NamedTemporaryFile
+    from zipfile import ZipFile
+
+    import requests
+
+    # globals
+    default_headers = {"Accept": "application/vnd.github.v3+json"}
+
+    # helper functions
+    def find_resource(resource, key, value):
+        for obj in resource:
+            if obj[key] == value:
+                return obj
+        return None
+
+    def handle_extract(file):
+        # if no extract is requested, exit here
+        if not extract:
+            return
+
+        dest_path = Path(__file__).parent / "InvenTree/web/static/web"
+
+        # if clean, delete static/web directory
+        if clean:
+            shutil.rmtree(dest_path, ignore_errors=True)
+            os.makedirs(dest_path)
+            print(f"Cleaned directory: {dest_path}")
+
+        # unzip build to static folder
+        with ZipFile(file, "r") as zip_ref:
+            zip_ref.extractall(dest_path)
+
+        print(f"Unzipped downloaded frontend build to: {dest_path}")
+
+    def handle_download(url):
+        # download frontend-build.zip to temporary file
+        with requests.get(url, headers=default_headers, stream=True, allow_redirects=True) as response, NamedTemporaryFile(suffix=".zip") as dst:
+            response.raise_for_status()
+
+            # auto decode the gzipped raw data
+            response.raw.read = functools.partial(response.raw.read, decode_content=True)
+            with open(dst.name, "wb") as f:
+                shutil.copyfileobj(response.raw, f)
+            print(f"Downloaded frontend build to temporary file: {dst.name}")
+
+            handle_extract(dst.name)
+
+    # if zip file is specified, try to extract it directly
+    if file:
+        handle_extract(file)
+        return
+
+    # check arguments
+    if ref is not None and tag is not None:
+        print("[ERROR] Do not set ref and tag.")
+        return
+
+    if ref is None and tag is None:
+        try:
+            ref = subprocess.check_output(["git", "rev-parse", "HEAD"], encoding="utf-8").strip()
+        except Exception:
+            print("[ERROR] Cannot get current ref via 'git rev-parse HEAD'")
+            return
+
+    if ref is None and tag is None:
+        print("[ERROR] Either ref or tag needs to be set.")
+
+    if tag:
+        tag = tag.lstrip("v")
+        try:
+            handle_download(f"https://github.com/{repo}/releases/download/{tag}/frontend-build.zip")
+        except Exception as e:
+            if not isinstance(e, requests.HTTPError):
+                raise e
+            print(f"""[ERROR] An Error occurred. Unable to download frontend build, release or build does not exist,
```

Comment:
just FYI, I used three concatenated strings as I wanted the indentation to be aligned, but its also fine like that if you prefer that way, Ill use it the next time.

Reply:
I find this way better understandable but no hard feelings either way